src/platforms/evdev-rs/src/lib.rs

```diff
@@ -1,25 +1,1404 @@
+                        device_info.event_builder = Some(unsafe {
+                            bridge_locked.create_event_builder_wrapper(event_builder.0.as_ptr())
+                        });
+                    }
+                }
+                ThreadCommand::GetPointerSettings(id, tx) => {
+                    if let Some(device_info) = find_device_by_id(state, id) {
+                        match device_info
+                            .device
+                            .has_capability(input::DeviceCapability::Pointer)
+                        {
+                            true => {
+                                let handedness = if device_info.device.config_left_handed() {
+                                    ffi::MirPointerHandedness::mir_pointer_handedness_left.repr
+                                } else {
+                                    ffi::MirPointerHandedness::mir_pointer_handedness_right.repr
+                                };
+
+                                let acceleration = if let Some(accel_profile) =
+                                    device_info.device.config_accel_profile()
+                                {
+                                    if accel_profile == input::AccelProfile::Adaptive {
+                                        ffi::MirPointerAcceleration::mir_pointer_acceleration_adaptive.repr
+                                    } else {
+                                        ffi::MirPointerAcceleration::mir_pointer_acceleration_none
+                                            .repr
+                                    }
+                                } else {
+                                    ffi::MirPointerAcceleration::mir_pointer_acceleration_none.repr
+                                };
+
+                                let acceleration_bias =
+                                    device_info.device.config_accel_speed() as f64;
+
+                                let settings = ffi::PointerSettingsRs {
+                                    is_set: true,
+                                    handedness: handedness,
+                                    cursor_acceleration_bias: acceleration_bias,
+                                    acceleration: acceleration,
+                                    horizontal_scroll_scale: state.x_scroll_scale,
+                                    vertical_scroll_scale: state.y_scroll_scale,
+                                };
+                                let _ = tx.send(settings);
+                            }
+                            false => {
+                                let settings = ffi::PointerSettingsRs::empty();
+                                let _ = tx.send(settings);
+                            }
+                        }
+                    } else {
+                        let settings = ffi::PointerSettingsRs::empty();
+                        let _ = tx.send(settings);
+                    }
```

Comment:
Here, too - asking for pointer settings on a non-pointer device should be an error? I suppose asking about a non-existent device would be a warning instead?